sklearn/preprocessing/_encoders.py

```diff
@@ -722,14 +747,26 @@ def get_feature_names_out(self, input_features=None):
         cats = self.categories_
         input_features = _check_feature_names_in(self, input_features)
 
+        name_combiner = self._get_feature_name_combiner()
         feature_names = []
         for i in range(len(cats)):
-            names = [input_features[i] + "_" + str(t) for t in cats[i]]
+            names = [name_combiner(input_features[i], t) for t in cats[i]]
             if self.drop_idx_ is not None and self.drop_idx_[i] is not None:
                 names.pop(self.drop_idx_[i])
             feature_names.extend(names)
         return np.asarray(feature_names, dtype=object)
 
+    def _get_feature_name_combiner(self):
+        if self.feature_name_combiner == "concat_string":
+            return lambda feature, category: feature + "_" + str(category)
+        elif callable(self.feature_name_combiner):
+            return self.feature_name_combiner
+        else:
+            raise ValueError(
+                "feature_name_combiner has to be either 'concat_string' or callable,"
+                " got {0}".format(self.feature_name_combiner)
```

Comment:
We are trying to move toward f-strings more:

```suggestion
                f" got {self.feature_name_combiner}"
```

Reply:
That's a good context, thanks! I wasn't sure since I saw a lot formatting calls around.